File: 01_Array_and_String/Python Code/03_1431. Kids With the Greatest Number of Candies.py
# Problem Link -> https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/description/

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        maxCandy = max(candies)
        res = []
        
        for i in candies:
            if i + extraCandies >= maxCandy:
                res.append(True)
            else:
                res.append(False)
        
        return res


# Time Complexity -> O(n)
# Space Complexity -> O(1)

File: 01_Array_and_String/C++ Code/03_1431. Kids With the Greatest Number of Candies.cpp
// Problem Link -> https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/description/

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> res;
        int maxCandy = *max_element(candies.begin(), candies.end());

        for(int i = 0; i < candies.size(); i++){
            if(candies[i] + extraCandies >= maxCandy){
                res.push_back(true);
            } else {
                res.push_back(false);
            }
        }

        return res;
    }
};



// Time Complexity -> O(n)
// Space Complexity -> O(1)
